bqskit/compiler/passes/partitioning/scan.py

```diff
@@ -0,0 +1,216 @@
+"""This module defines the ScanPartitioner pass."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from bqskit.compiler.basepass import BasePass
+from bqskit.compiler.machine import MachineModel
+from bqskit.ir.circuit import Circuit
+from bqskit.ir.region import CircuitRegion
+from bqskit.utils.typing import is_integer
+
+_logger = logging.getLogger(__name__)
+
+
+class ScanPartitioner(BasePass):
+    """
+    The ScanPartitioner Pass.
+
+    This pass forms partitions in the circuit by scanning from left
+    to right. This is based on the partitioner from QGo.
+
+    References:
+        Wu, Xin-Chuan, et al. "QGo: Scalable Quantum Circuit Optimization
+        Using Automated Synthesis." arXiv preprint arXiv:2012.09835 (2020).
+    """
+
+    def __init__(
+        self,
+        block_size: int = 3,
+        single_gate_score: int = 1,
+        multi_gate_score: int = 1000,  # TODO: Pass callable scoring_fn instead
+    ) -> None:
+        """
+        Construct a SimplePartitioner.
```

Comment:
```suggestion
        Construct a ScanPartitioner.
```

Also do we want to call this the QGoPartitioner pass?

Reply:
No, QGO refers to the entire partitioning-synthesis compilation algorithm. We should decouple that from the partitioner.